relayer/src/ethereum_checkpoints/replay_back.rs

```diff
@@ -1,15 +1,29 @@
 use super::*;
 use ethereum_beacon_client::{self, BeaconClient};
 
+pub struct ExecutionArgs {
+    pub beacon_client: BeaconClient,
+    pub client: GearApi,
+    pub program_id: [u8; 32],
+    pub gas_limit: u64,
+    pub replay_back: Option<ReplayBack>,
+    pub checkpoint: (Slot, Hash256),
+    pub sync_update: SyncCommitteeUpdate,
+    pub size_batch: u64
+}
+
 #[allow(clippy::too_many_arguments)]
 pub async fn execute(
-    beacon_client: &BeaconClient,
-    client: &GearApi,
-    program_id: [u8; 32],
-    gas_limit: u64,
-    replay_back: Option<ReplayBack>,
-    checkpoint: (Slot, Hash256),
-    sync_update: SyncCommitteeUpdate,
+    ExecutionArgs {
+        beacon_client,
+        client,
+        program_id,
+        gas_limit,
+        replay_back,
+        checkpoint,
+        sync_update,
+        size_batch,
+    }: ExecutionArgs,
```

Comment:
Many sources criticize that approach since the function signature becomes too noisy. So I ask you to make de-structuring within the function body